app/helpers/friendships_helper.rb

```diff
@@ -46,4 +23,35 @@ def get_friendship(user)
     relation = Friendship.find_by(sender:user.id, receiver:current_user.id) || 
                Friendship.find_by(sender:current_user.id, receiver:user.id) 
   end
+
+  def get_all_friends_id(user)
+    friends = Friendship.select{|f| f.status == true and f.sender == user.id}
+    @friends_id = []
+    friends.each do |f|
+      @friends_id << f.sender
+      @friends_id << f.receiver
+    end
+    @friends_id = @friends_id == [] ? user.id : @friends_id.flatten.uniq
+  end
+
+  def pending_friendship(user)
+    friends = []
+
+    user.receivers.each do |receiver|
+      friends << User.find_by(id: receiver.sender) if receiver.status == false
+    end
+
+    friends
+  end
+
+  def get_friends(user)
+    friends = []
+
+    user.senders.each do |sender|
+      friends << User.find_by(id: sender.receiver) if sender.status == true
```

Comment:
Not supposed to have application logic in your helper files. `find_by` methods are application logic related.